Add virtualenv removal events

matan129 opened the pull request:
Hi,
For my own use, I'm intersted in virtualenv destruction events, so I added them.

Basically I want to be able to register custom Jupyter kernels per venv,
and to be able to remove them automatically when the venv is deleted.
I didn't include my [implementaion](https://gist.github.com/matan129/a95b2bc11086df0ffe7491b4149f9783) of that, since it's very opinionated, I think.

Thanks

justinmayer commented:
Hi Matan. Thanks for the contribution. I think perhaps these events might be better called `virtualenv_will_remove` and `virtualenv_did_remove`, just to keep consistent with the corresponding existing terminology.

Regarding the help-text spacing, that's probably best handled outside the context of this pull request. That said, I'm not sure I want to push the longest line from 85 to 90 characters, so I think for the moment it's probably best to just remove that particular commit from this PR.

matan129 commented:
I've renamed the events.
Just before you've commented, I changed the spacing so it will be dynamic, based on what subcommands are loaded.
So in effect it will have not impact on the current commands, only when someone (like me) adds a particulary long command (for me, it's unregister_jupyter).

matan129 commented:
By the way, is there any reason this project is purely fish?
(Using [fisher](https://github.com/jorgebucaran/fisher), for instance)

justinmayer commented:
Thanks, Matan. I'll have a look at the dynamic spacing code. As for the latter question, I think that topic is already well covered by the discussion in #109.

justinmayer commented on virtualfish/virtual.fish:
```diff
@@ -440,9 +442,20 @@ function __vf_help --description "Print VirtualFish usage information"
     echo "Available commands:"
     echo
-    for sc in (functions -a | sed -n '/__vf_/{s///g;p;}')
+
```
Perhaps put a comment just above this stanza that explains its purpose? Something like:

    # Dynamically calculate column spacing, based on longest subcommand

justinmayer commented on virtualfish/virtual.fish:
```diff
@@ -440,9 +442,20 @@ function __vf_help --description "Print VirtualFish usage information"
+            set max_subcommand_length $cur
+        end
+    end
+    set -l spacing (math $max_subcommand_length + 2)
```
> So in effect it will have no impact on the current commands

In my testing, this yielded three spaces between the longest subcommand (`globalpackages`) and the description column, whereas the gap was previously only two spaces. Not a big deal, obviously, but is there any harm in changing this line to: `set -l spacing (math $max_subcommand_length + 1)` ?

matan129 commented:
Done :)

justinmayer commented:
Fantastic. Many thanks for the contribution, Matan! ✨

justinmayer commented:
These events never got documented, so I added them to the event list and made some minor standardization improvements to the new removal events via 420b38b, all included in the [just-released v2.5](https://github.com/justinmayer/virtualfish/releases/tag/2.5.0).
